TypeScript: DrawerTypes.TEMPORARY

TypeScript definition does not correctly convey that `Drawer.DrawerTypes.TEMPORARY` exists:

```
error TS2339: Property 'TEMPORARY' does not exist on type 'DrawerTypes'.
  Property 'TEMPORARY' does not exist on type '{ FULL_HEIGHT: "full-height"; CLIPPED: "clipped"; FLOATING: "floating"; PERSISTENT: "persistent";...'.
```

--- a/index.d.ts
+++ b/index.d.ts
@@ -32,7 +32,7 @@ export {
   Dialog, DialogProps,
 } from './lib/Dialogs';
 export { default as Divider, DividerProps } from './lib/Dividers';
-export { default as Drawer, DrawerProps, MobileDrawerTypes, DrawerTypes } from './lib/Drawers';
+export { default as Drawer, DrawerProps, MobileDrawerType, DrawerType } from './lib/Drawers';
 export {
   ExpansionPanel, ExpansionPanelProps,
   ExpansionList, ExpansionListProps,

--- a/src/js/Drawers/Drawer.d.ts
+++ b/src/js/Drawers/Drawer.d.ts
@@ -1,28 +1,11 @@
 import * as React from 'react';
 import { Props } from '../index';
 
-export type MobileDrawerTypes = {
-  // Temporary
-  TEMPORARY: 'temporary',
-  TEMPORARY_MINI: 'temporary-mini',
-}
+export type MobileDrawerType = 'temporary' | 'temporary-mini';
+export type DrawerType = 'full-height' | 'clipped' | 'floating' | 'persistent' | 'persistent-mini' | MobileDrawerType;
 
-export type DrawerTypes = {
-  // Permanent drawers
-  FULL_HEIGHT: 'full-height',
-  CLIPPED: 'clipped',
-  FLOATING: 'floating',
-
-  // Persistent drawers
-  PERSISTENT: 'persistent',
-  PERSISTENT_MINI: 'persistent-mini',
-} | MobileDrawerTypes;
-
-export type MobileDrawerTypesType = 'temporary' | 'temporary-mini';
-export type DrawerTypesType = 'full-height' | 'clipped' | 'floating' | 'persistent' | 'persistent-mini' | MobileDrawerTypesType;
-
-export type MediaTypes = 'mobile' | 'tablet' | 'desktop';
-export type DrawerPositions = 'left' | 'right';
+export type MediaType = 'mobile' | 'tablet' | 'desktop';
+export type DrawerPosition = 'left' | 'right';
 
 export interface DrawerProps extends Props {
   // for the `component` prop until refactored out
@@ -36,23 +19,23 @@ export interface DrawerProps extends Props {
   navItems?: Array<React.ReactElement<any> | { divider?: boolean, subheader?: boolean, primaryText?: string }>;
   autoclose?: boolean;
   header?: React.ReactNode;
-  mobileType?: MobileDrawerTypes | MobileDrawerTypesType;
+  mobileType?: MobileDrawerType;
   mobileMinWidth?: number;
-  tabletType?: DrawerTypes | DrawerTypesType;
+  tabletType?: DrawerType;
   tabletMinWidth?: number;
-  desktopType?: DrawerTypes | DrawerTypesType;
+  desktopType?: DrawerType;
   desktopMinWidth?: number;
-  type?: DrawerTypes;
-  onMediaTypeChange?: (type: DrawerTypesType, media: { mobile: boolean, tablet: boolean, desktop: boolean }) => void;
+  type?: DrawerType;
+  onMediaTypeChange?: (type: DrawerType, media: { mobile: boolean, tablet: boolean, desktop: boolean }) => void;
   onVisibilityChange?: (visible: boolean, event: React.MouseEvent<HTMLElement>) => void;
-  defaultMedia?: MediaTypes;
+  defaultMedia?: MediaType;
   overlay?: boolean;
   portal?: boolean;
   renderNode?: Object;
   lastChild?: boolean;
   defaultVisible?: boolean;
   visible?: boolean;
-  position?: DrawerPositions;
+  position?: DrawerPosition;
   inline?: boolean;
   transitionDuration?: number;
   clickableDesktopOverlay?: boolean;
@@ -84,10 +67,10 @@ interface DrawerComponent extends React.ComponentClass<DrawerProps> {
     mobileMinWidth: number,
     tabletMinWidth: number,
     desktopMinWidth: number,
-    mobileType: MobileDrawerTypes | MobileDrawerTypesType,
-    tabletType: DrawerTypes | DrawerTypesType,
-    desktopType: DrawerTypes | DrawerTypesType,
-  }): { type: DrawerTypes | DrawerTypesType, mobile: boolean, tablet: boolean, desktop: boolean };
+    mobileType: MobileDrawerType,
+    tabletType: DrawerType,
+    desktopType: DrawerType,
+  }): { type: DrawerType, mobile: boolean, tablet: boolean, desktop: boolean };
 
   matchesMedia(min: number, max?: number): boolean;
 }

--- a/src/js/Drawers/index.d.ts
+++ b/src/js/Drawers/index.d.ts
@@ -1,10 +1,8 @@
 export {
   default,
   DrawerProps,
-  MobileDrawerTypes,
-  MobileDrawerTypesType,
-  DrawerTypes,
-  DrawerTypesType,
-  MediaTypes,
-  DrawerPositions,
+  MobileDrawerType,
+  DrawerType,
+  MediaType,
+  DrawerPosition,
 } from './Drawer';

--- a/src/js/Helpers/Layover.d.ts
+++ b/src/js/Helpers/Layover.d.ts
@@ -18,13 +18,7 @@ export type VerticalAnchors = {
   BOTTOM: 'bottom'
 }
 
-export type LayoverPositions =  {
-  TOP_LEFT: 'tl',
-  TOP_RIGHT: 'tr',
-  BOTTOM_LEFT: 'bl',
-  BOTTOM_RIGHT: 'br',
-  BELOW: 'below'
-}
+export type LayoverPositions = 'tl' | 'tr' | 'bl' | 'br' | 'below';
 
 export interface LayoverAnchor {
   x: HorizontalAnchors;

--- a/src/js/NavigationDrawers/NavigationDrawer.d.ts
+++ b/src/js/NavigationDrawers/NavigationDrawer.d.ts
@@ -1,12 +1,10 @@
 import * as React from 'react';
 import { Props } from '../index';
 import {
-  DrawerTypes,
-  DrawerTypesType,
-  MobileDrawerTypes,
-  MobileDrawerTypesType,
-  DrawerPositions,
-  MediaTypes,
+  DrawerType,
+  MobileDrawerType,
+  DrawerPosition,
+  MediaType,
 } from '../Drawers';
 
 export interface NavigationDrawerProps extends Props {
@@ -26,20 +24,20 @@ export interface NavigationDrawerProps extends Props {
   drawerHeaderChildren?: React.ReactNode;
   drawerTitle?: React.ReactNode;
   drawerChildren?: React.ReactNode;
-  position?: DrawerPositions;
+  position?: DrawerPosition;
   navItems?: Array<React.ReactElement<any> | { divider?: boolean, subheader?: boolean, primaryText?: React.ReactNode }>;
-  mobileDrawerType?: MobileDrawerTypes | MobileDrawerTypesType;
-  tabletDrawerType?: DrawerTypes | DrawerTypesType;
-  desktopDrawerType?: DrawerTypes | DrawerTypesType;
-  drawerType?: DrawerTypes | DrawerTypesType;
-  defaultMedia?: MediaTypes;
+  mobileDrawerType?: MobileDrawerType;
+  tabletDrawerType?: DrawerType;
+  desktopDrawerType?: DrawerType;
+  drawerType?: DrawerType;
+  defaultMedia?: MediaType;
   mobileMinWidth?: number;
   tabletMinWidth?: number;
   desktopMinWidth?: number;
   portal?: boolean;
   lastChild?: boolean;
   renderNode?: Object;
-  onMediaTypeChange?: (type: DrawerTypesType, media: { mobile: boolean, tablet: boolean, desktop: boolean }) => void;
+  onMediaTypeChange?: (type: DrawerType, media: { mobile: boolean, tablet: boolean, desktop: boolean }) => void;
   defaultVisible?: boolean;
   visible?: boolean;
   onVisibilityChange?: (visible: boolean, event: Event) => void;
@@ -113,10 +111,10 @@ interface NavigationDrawerComponent extends React.ComponentClass<NavigationDrawe
     mobileMinWidth: number,
     tabletMinWidth: number,
     desktopMinWidth: number,
-    mobileDrawerType: MobileDrawerTypes | MobileDrawerTypesType,
-    tabletDrawerType: DrawerTypes | DrawerTypesType,
-    desktopDrawerType: DrawerTypes | DrawerTypesType,
-  }): { type: DrawerTypesType | DrawerTypes, mobile: boolean, tablet: boolean, desktop: boolean };
+    mobileDrawerType: MobileDrawerType,
+    tabletDrawerType: DrawerType,
+    desktopDrawerType: DrawerType,
+  }): { type: DrawerType, mobile: boolean, tablet: boolean, desktop: boolean };
 }
 
 declare const NavigationDrawer: NavigationDrawerComponent;

--- a/src/js/Tabs/Tabs.d.ts
+++ b/src/js/Tabs/Tabs.d.ts
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { IdPropType, Props } from '../index';
 import { TabProps as Tab } from './Tab';
-import { MediaTypes } from '../Drawers';
+import { MediaType } from '../Drawers';
 
 export interface TabsProps extends Props {
   // for the `component` prop until refactored out
@@ -23,7 +23,7 @@ export interface TabsProps extends Props {
   ) => void;
   activeTabIndex?: number;
   defaultTabIndex?: number;
-  defaultMedia?: MediaTypes;
+  defaultMedia?: MediaType;
   desktopMinWidth?: number;
   nextIcon?: React.ReactElement<any>;
   previousIcon?: React.ReactElement<any>;

--- a/src/js/Tabs/TabsContainer.d.ts
+++ b/src/js/Tabs/TabsContainer.d.ts
@@ -1,6 +1,5 @@
 import * as React from 'react';
 import { IdPropType, Props, BaseProps } from '../index';
-import { MediaTypes } from '../Drawers';
 import { TabsProps as Tabs } from './Tabs';
 
 export interface TabsContainerProps extends Props {

